### ISSUE TITLE
Issues when publishing to Sonatype Maven

### ISSUE THREAD

**Issue by flelli** [mooltiverse/nyx#9]
This is actually a documentation issue authored afterwards but since publishing to [Sonatype Maven](https://oss.sonatype.org/) often gives issues it's worth to leave a note.

Most of the configuration is done in the [`build.gradle` script within the `modules/java` directory](https://github.com/mooltiverse/nyx/blob/master/modules/java/build.gradle).

Publishing to [Sonatype Maven](https://oss.sonatype.org/) is an intermediate step to have artifacts published to [Maven Central](https://search.maven.org/) but it requires a tricky process and configuration, carried out by the plugins configured and documented in the [`build.gradle` script](https://github.com/mooltiverse/nyx/blob/master/modules/java/build.gradle).

However, although everything is configured properly, publishing often still fails. When this happens you can:
* try again later: Sonatype Nexus has serious performance issues and takes a lot of time to perform its tasks. Sometimes it hangs the connection, other times it just can't complete in the configured timeout (consider the script already has a huge number of retries configured: 300). It often happens that retrying after a while just gets it back to work
* check the staging repository status and make sure there is no stale repository (see below for more)

## Sonatype Nexus staging repositories
Apart from all the process behind it, when we publish to Maven we actually create a new staging repository onto Sonatype Maven [Sonatype Maven](https://oss.sonatype.org/) and once the verification process completes succesfully its content is moved to [Maven Central](https://search.maven.org/).
In other words, we never publish directly to Maven Central but only to some temporary staging repository to Sonatype Maven. Each publication creates a new, temporary, staging repository that will be then deleted automatically upon completion.

When the process breaks for some reason a staging repository may remain stale and that sometimes prevents further publications to succeed.

To solve this issue we have to:
* manually connect to [Sonatype Maven](https://oss.sonatype.org/)
* log in with the project service account
* go to Staging Repositories (on the left side menu) to see the current staging repositories
* delete the stale repositories, if any
